cmd/flipt/validate.go

```diff
@@ -37,10 +45,48 @@ func newValidateCommand() *cobra.Command {
 }
 
 func (v *validateCommand) run(cmd *cobra.Command, args []string) {
-	if err := cue.ValidateFiles(os.Stdout, args, v.format); err != nil {
-		if errors.Is(err, cue.ErrValidationFailed) {
+	validator, err := cue.NewFeaturesValidator()
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	for _, arg := range args {
+		f, err := os.ReadFile(arg)
+		if err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+
+		res, err := validator.Validate(arg, f)
+		if err != nil && !errors.Is(err, cue.ErrValidationFailed) {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+
+		if len(res.Errors) > 0 {
+			if v.format == jsonFormat {
+				_ = json.NewEncoder(os.Stdout).Encode(res)
+				os.Exit(v.issueExitCode)
+				return
+			}
+
+			var sb strings.Builder
+			sb.WriteString("❌ Validation failure!\n")
+
+			for _, e := range res.Errors {
+				msg := fmt.Sprintf(`
+- Message  : %s
+	File     : %s
+	Line     : %d
+	Column   : %d
+`, e.Message, e.Location.File, e.Location.Line, e.Location.Column)
+
+				sb.WriteString(msg)
+			}
+
+			fmt.Println(sb.String())
```

Comment:
Take it or leave it:

I don't think the string builder buys us anything, as we allocate a new one each time and then read the contents out and have to then copy it onto stdout writer. Might as-well write straight to stdout.
```suggestion
			fmt.Println("❌ Validation failure!")

			for _, e := range res.Errors {
				fmt.Printf(`
- Message  : %s
	File     : %s
	Line     : %d
	Column   : %d
`, e.Message, e.Location.File, e.Location.Line, e.Location.Column)
			}
```